### PR TITLE
[fix] add secure rm

### PR DESCRIPTION
**Pull request by magikcypress**
Adding a secure rm function to avoid doing a brutal rm

### CODE REVIEW

File: scripts/install [hunk 1]
```diff
@@ -27,14 +28,17 @@ final_path=/var/www/$app
...
+# Files owned by user specific can just read
 sudo find $final_path -type f | xargs sudo chmod 644
 sudo find $final_path -type d | xargs sudo chmod 755
-sudo chown -R root: $final_path
+sudo chown -R $app: $final_path
 
 # except for data and tmp subdir, where www-data must have write permissions
 sudo mkdir -p $final_path/{data,tmp}
-sudo chown -R www-data:root $final_path/{data,tmp}
+sudo chown -R $app:root $final_path/{data,tmp}
```
Comment (maniackcrudelis) [inline scripts/install]: If you need write permission in those directories. I guess you don't need it on another directories.
So keep, at the line 37, root as owner of the main directory.
Your specific user have a read access, it's enough.

Comment (magikcypress) [inline scripts/install]: Like this ?

`sudo chown -R $app: $final_path/{data,tmp}`

Comment (maniackcrudelis) [inline scripts/install]: No it's worse, just don't change the line 37.
Keep root as owner.

**Comment by JimboJoe** [comment]
Still some problems on file rights...
> `2017/04/23 06:41:20 [error] 25532#0: *472 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'Exception' with message 'unable to write to file data/.htaccess' in /var/www/zerobin/lib/Persistence/AbstractPersistence.php:104
Stack trace:
#0 /var/www/zerobin/lib/Persistence/AbstractPersistence.php(73): PrivateBin\Persistence\AbstractPersistence::_initialize()
#1 /var/www/zerobin/lib/Persistence/TrafficLimiter.php(106): PrivateBin\Persistence\AbstractPersistence::_exists('traffic_limiter...')
#2 /var/www/zerobin/lib/PrivateBin.php(223): PrivateBin\Persistence\TrafficLimiter::canPass()
#3 /var/www/zerobin/lib/PrivateBin.php(135): PrivateBin\PrivateBin->_create()
#4 /var/www/zerobin/index.php(18): PrivateBin\PrivateBin->__construct()
#5 {main}
  thrown in /var/www/zerobin/lib/Persistence/AbstractPersistence.php on line 104" while reading response header from upstream, client: 10.0.2.2, server: ortest.noho.st, request: "POST /zerobin/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "ortest.noho.st"
`

**Comment by JimboJoe** [comment]
@magikcypress if I understand correctly, we're waiting for [this](https://github.com/PrivateBin/PrivateBin/pull/226#discussion_r114010427) to get finalized and merged in PrivateBin, right?

**Comment by magikcypress** [comment]
Currently, I have to work on a patch on PrivateBin, but I don't have time to make it. This patch does not write the .htaccess for Nginx. But, from what I understand @JimboJoe problem is not related to this .htaccess problem. This is a problem between php5-fpm and the dedicated user, i think more research should be done.

**Comment by JimboJoe** [comment]
I made some changes so that package_check results look much better now.
That still needs some extensive testing before we can release it and create the privatebin_ynh package...
Some volunteers? :-)
(I don't use that app myself)

**Comment by magikcypress** [comment]
I worked a little bit ;)

https://github.com/magikcypress/PrivateBin/commit/f23b1222dd8e296858a45384185c2175a0aaf7cd

**Comment by maniackcrudelis** [comment]
Restore script failed !
You should remove this useless lines
https://github.com/YunoHost-Apps/zerobin_ynh/blob/devel/scripts/restore#L76-L78

**Comment by alexAubin** [comment]
Bumping while passing by ? This PR is needed for https://github.com/YunoHost/apps/pull/212

**Comment by JimboJoe** [comment]
I think this PR should be merged right now and the decision process applied to the whole application...

**Comment by maniackcrudelis** [comment]
If the work on this app is over. I agree.
And we report the decision process on the PR for the app list.

**Comment by magikcypress** [comment]
I agree with everyone ;)

**Comment by Rafi594** [comment]
Works fine on my server

**Comment by alexAubin** [comment]
(looks like there's a readme conflict to solve ;))

**Comment by M5oul** [comment]
In fact, I merged the other PR with readme modifications.
I resolved the conflict.

**Pull request merged**